gateway/mw_transform_test.go

```diff
@@ -349,3 +350,41 @@ func TestBodyTransformCaseSensitivity(t *testing.T) {
 		assert("/Get", "/Get", `{"http_method":"GET"}`)
 	})
 }
+
+func TestTransformRequestBody(t *testing.T) {
```

Comment:
What if instead you unit test the APIDefinitionLoader `compileTransformPathSpec`; we need more unit tests and less integration tests, and this is a case that is unit testable...